docs/README.md

```diff
@@ -1,9 +1,37 @@
 # Documentation
 
-## [Code of Conduct](../CODE_OF_CONDUCT.md)
+## Terminology
+There can be a lot of confusion what an `app` is. It can mean a server app conventionally written in PHP, the app implementation of one or a mobile app built with Neon.
+Therefore we use the following naming:
+- **App** as the resulting shipped apped enabling one or more clients
+- **Client** for the implementation of a server PHP app
+- **Plugin** the implementations of server functions that are used by multiple clients
+- **Nextcloud-Client** as the Dart client generated from the OpenAPI specifications
+- **Server app** as the app running on the server conventionally built with PHP
+- **Neon project** is this entire mono repository containing multiple packages
+- **Neon framework** the Flutter Nexrtcloud app framework itself
 
-## [Contributing](../CONTRIBUTING.md)
+## Code of Conduct 
+Please read the [Code of Conduct]((../CODE_OF_CONDUCT.md)). This document offers some guidance to ensure Nextcloud and Neon participants can cooperate effectively in a positive and inspiring atmosphere, and to explain how together we can strengthen and support each other.
 
-## [Architecture overview](architecture.md)
+## Contributing
+Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository to help you get started with our workflows and tools.
+
+## Architecture overview 
+Check out our little [overview](architecture.md) of how the Neon project is built to help you get started.
 
 ## [Login user flow](login.md)
+
+## Nextcloud Conference 2023
+We held a lightning talk in the at the Nextcloud conference in 2023 in Berlin.
+You cen check our [slides](nextcloud_conference_talks.pdf) and the recording once available.
+
+## Adding a new package to the Neon project
```

Comment:
I think this should either go into the architecture documentation or be its own page.